query-planner-js/src/buildQueryPlan.ts

```diff
@@ -688,6 +689,24 @@ function splitFields(
   }
 }
 
+function shouldIncludeToFetchGroup(
+  group: FetchGroup,
+  fields: FieldSet,
+  context: QueryPlanningContext,
+  runtimeParentType: GraphQLObjectType
+  ): Boolean {
+  const { fieldDef } = fields[0];
+  const returnType = getNamedType(fieldDef.type);
+
+  if (isCompositeType(returnType)) {
```

Comment:
Curious as to why you added this exclusion? I don't mind keeping it as it limits the cases changed by this fix and is thus somewhat more conservative, but given the goal is to exclude fields from services that don't the field at all, I'm not sure to see why the field return type would influence the behaviour.

Mostly wondering if you were thinking of a case I don't have in mind.

Reply:
Yes, that was the intention, to make changes as little as possible, because I'm not very familiar with the query planner.